polaris-service/src/main/java/org/apache/polaris/service/config/PolarisApplicationConfig.java

```diff
@@ -208,4 +215,58 @@ public void setAwsSecretKey(String awsSecretKey) {
   public void setDefaultRealms(List<String> defaultRealms) {
     this.defaultRealms = defaultRealms;
   }
+
+  public Supplier<GoogleCredentials> getGcpCredentialsProvider() {
+    return gcpAccessToken != null
+        ? () -> GoogleCredentials.create(gcpAccessToken)
+        : () -> {
+          try {
+            return GoogleCredentials.getApplicationDefault();
+          } catch (IOException e) {
+            throw new RuntimeException("Failed to get GCP credentials", e);
+          }
+        };
+  }
+
+  @JsonProperty("gcp_credentials")
+  void setGcpCredentials(GcpAccessToken token) {
+    this.gcpAccessToken =
+        new AccessToken(
+            token.getAccessToken(),
+            new Date(System.currentTimeMillis() + token.getExpiresIn() * 1000));
+  }
+
+  /**
+   * A static AccessToken representation used to store a static token and expiration date. This
+   * should strictly be used for testing.
```

Comment:
If it's only used for testing, can we not put it in `test`?

Reply:
No, we need a type to set on the setter method parameter, so something needs to be visible here. If the type was defined in test, it wouldn't be visible in this class

Reply:
> we need a type to set on the setter method parameter

We need a type in `main`, but it could be an interface right? I think it's better not to have code that should only be used in tests outside of `test`

Reply:
There's a pretty substantial number of classes that are only used in tests - e.g., the DefaultContextResolver is only for tests, as is the in-memory metastore manager.

TBH, I _do_ really want to have better support for customizable cloud-specific credential loading - e.g., read the GCP credentials from vault or assume an AWS IAM role using an instance profile - but adding support for that is more than just defining an interface. We have to design an interface that subclasses `Discoverable`, hash out the details around that interface, then add the `META-INF/services` files in the `main` and `test` structures, then override the integration-test yml file to specify which one to use and... that's more work than I think ought to go into this bugfix. 

I think the barrier to adding a test for a bugfix should be low. We can refactor to better support for injectable credential providers later, but I don't think it makes sense to hamstring a bugfix PR because we didn't properly abstract this code in the first place. I will happily create an issue and add a TODO in the code, but I think fixing the bug is more important to get done

Reply:
> the DefaultContextResolver is only for tests, as is the in-memory metastore manager.

I see, so similar to these types, is it accurate to say that this implementation can be used for _manual testing_, not just _automated tests_? I originally understood this comment's "testing" to refer just to automated tests. Sorry about that.

Reply:
The main purpose is for tests, but users could use it for manual testing if they really needed it.

If your concern is the visibility of the class itself, it is a package-private class, so isn't visible outside of that package. I can certainly make it private instead, if that alleviates your concern.

Reply:
I see, if it's used for manual testing then I think it's okay for leave it here for now and refactor later. Thanks for clarifying!